app/utils/write_file/write-file.js

```diff
@@ -55,19 +65,20 @@ async function writeJpegToDisk(base64, fileName) {
  */
 async function writePngToDisk(base64, fileName) {
     if (base64 && fileName) {
-        let writeFile = new Promise((resolve) => {
-            const buffer = new Buffer(base64, 'base64')
-            pngToJpeg({quality: 90})(buffer).then((output) => {
-                fs.writeFile(fileName, output, (error) => {
-                    if (error) throw error.internalError()
-                    else resolve()
-                })
-            }).catch((error) => {
-                if (error instanceof Error) throw error
-                else throw returnError.internalError()
+        const buffer = new Buffer(base64, 'base64')
+        await pngToJpeg({quality: 90})(buffer).then((output) => {
+            fs.writeFile(fileName, output, (error) => {
+                if (error) {
+                    logger.log('error', error)
+                    throw error.internalError()
+                }
```

Comment:
`await` isn't actually doing anything here. You're treating the promise like a regular promise by using `then` and `catch`. The point of `await` is to synchronously run the promise object. You want to use an await try-catch approach here

```javascript
try { 
    const output = await pngToJpeg({quality: 90})(buffer)
    // the fs.writeFile piece must be promise-fied
    await writeFile(filename, output)
} catch (err) {
   if (err instanceof Error) {
   } else throw returnError.internalError()
}

return filename
```

```javascript
function writeFile(filename, output) {
   return new Promise((resolve, reject) => {
      fs.writeFile(filename, output, (error) => {
          if (error) {
                logger.log('error', error)
                reject(error.internalError())
          }
          else resolve()
      })
   })
}
```

This will roughly do what you're trying to accomplish. Syntax might need to be corrected for